dmoj/cptbox/isolate.py

```diff
@@ -314,63 +348,97 @@ def _file_access_check(
         # This works, except when the child process uses /proc/self, which refers to something else in this process.
         # Therefore, we "project" it by changing it to /proc/[tid] for computing the realpath and doing the samefile
         # check. However, we still keep it as /proc/self when checking access rules.
+
+        # normpath doesn't strip leading slashes
         projected = normalized = '/' + os.path.normpath(file).lstrip('/')
+        proc_dir = f'/proc/{debugger.tid}'
         if normalized.startswith('/proc/self'):
-            file = os.path.join(f'/proc/{debugger.tid}', os.path.relpath(file, '/proc/self'))
-            projected = '/' + os.path.normpath(file).lstrip('/')
-        elif normalized.startswith(f'/proc/{debugger.tid}/'):
+            projected = os.path.join(proc_dir, os.path.relpath(projected, '/proc/self'))
+        elif normalized.startswith(proc_dir):
```

Comment:
This changes behavior to no longer require the trailing slash

Reply:
It can generate some non-normalized paths now:

```python

In [5]: proc_dir = '/proc/123'

In [6]: normalized = '/proc/123'

In [7]: os.path.join('/proc/self', os.path.relpath(normalized, proc_di
   ...: r))
Out[7]: '/proc/self/.'
```

Reply:
Is it better to normalize, or to include the trailing slash? Supposedly we don't want to check the literal path `/proc/self` wrong, since under the old system it won't match

Reply:
I'd rather not have to think about this subtle of a change within this big of a PR, if you'd like to make a change to this logic please wait until afterwards to do so.